### PR TITLE
Add a common authorization module

### PR DESCRIPTION
**Pull request by cveneziani**

### CODE REVIEW

File: apps/web/controllers/trips/authorizations/base.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module Web::Trips
+  module Authorizations
```
Comment (nfilzi) [inline apps/web/controllers/trips/authorizations/base.rb]: So, are we not going full Hanami style on these module definitions?
Like 
```
module Web::Trips::Authorizations
  class Update
  end
end
```

File: apps/web/controllers/trips/authorizations/base.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      def future_trip?
+        trip.starting_on >= Date.today
+      end
+
+      def user_organizer?
+        TripOrganizerRepository.new.organizes_trip?(user.id, trip.id)
+      end
+
+      def user_signed_in?
+        !!user
```
Comment (nfilzi) [inline apps/web/controllers/trips/authorizations/base.rb]: Aha I learnt about this little trick a couple of weeks ago 😂 👌

File: apps/web/controllers/trips/housings/authorizations/create.rb [hunk 3]
```diff
@@ -2,6 +2,8 @@ module Web::Trips
   module Housings
     module Authorizations
       class Create
+        include Web::Trips::Authorizations::Base
```
Comment (nfilzi) [inline apps/web/controllers/trips/housings/authorizations/create.rb]: Why not use the `::` at the start this time? Any particular reason not to have `include ::Web::Trips::Authorizations::Base` ?

Comment (cveneziani) [reply apps/web/controllers/trips/housings/authorizations/create.rb]: We can.

File: apps/web/controllers/trips/authorizations/base.rb (hunk 1 above)
Comment (cveneziani) [reply apps/web/controllers/trips/authorizations/base.rb]: We can if you want.

Comment (nfilzi) [reply apps/web/controllers/trips/authorizations/base.rb]: Let's do it, except if there is a caveat I'm not aware of when using this style of namespacing..?

File: apps/web/controllers/trips/housings/authorizations/create.rb (hunk 3 above)
Comment (nfilzi) [reply apps/web/controllers/trips/housings/authorizations/create.rb]: Let's do it.

File: apps/web/controllers/trips/authorizations/base.rb (hunk 1 above)
Comment (cveneziani) [reply apps/web/controllers/trips/authorizations/base.rb]: Let's try!

**Comment by cveneziani** [comment]
Fixed.

**Comment by nfilzi** [comment]
Already approved 😬

**Pull request merged**